packages/qdrant-loader/src/qdrant_loader/connectors/git/connector.py

```
@@ -432,6 +436,40 @@ async def get_documents(self) -> list[Document]:
+        if file_path is None:
+            self.logger.warning(
+                "Path traversal attempt blocked", entity_id=entity_id
+            )
+            return None
+        if not os.path.exists(file_path) or not self.file_processor.should_process_file(  # type: ignore
+            file_path
+        ):
+            return None
+        try:
+            return self._process_file(file_path)
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)